frontend/src/components/data-table/focus-row/types.ts

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+/* eslint-disable @typescript-eslint/no-empty-interface */
+
+import type { OnChangeFn, RowData } from "@tanstack/react-table";
+
+// define types for feature's custom state
+export type FocusRowState = number;
+export interface FocusRowTableState {
+  focusedRowIdx: FocusRowState;
+}
+
+// define types for feature's table options
+export interface FocusRowOptions {
+  enableFocusRow?: boolean;
+  onFocusRowChange?: OnChangeFn<FocusRowState>;
+}
+
+// Define types for feature's table APIs
+export interface FocusRowInstance {
+  focusRow: (rowIdx: FocusRowState) => void;
+  getFocusedRowIdx: () => FocusRowState;
```

Comment:
These need to be optional since the feature is optional. Otherwise it might break in the variable sidebar